File: Lab2krimal/app/src/main/java/fall2020/mobileapp/lab2_krimal/MainActivity.java
package fall2020.mobileapp.lab2_krimal;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    int excellent_tip = 20;
    int average_tip = 18;
    int bad_tip = 14;
    Button button;
    Button Lab1Button;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        /*Intent intent = getIntent();

        excellent_tip = intent.getIntExtra("NUMBER1", excellent_tip);
        average_tip= intent.getIntExtra("NUMBER2", average_tip);
        bad_tip = intent.getIntExtra("NUMBER3", bad_tip);*/


        button = (Button) findViewById(R.id.secondActivityButton);
        Lab1Button = (Button) findViewById(R.id.Implicit_call);


        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openNewActivity();

            }
        });

        Lab1Button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Lab1Activity();
            }
        });

    }

    public void Lab1Activity(){
        Intent intent = new Intent();
        intent.setAction("android.intent.action.VIEW");
        intent.addCategory("android.intent.category.DEFAULT");
        startActivity(intent);
    }

    public void openNewActivity(){
        Intent intent = new Intent(this, NewActivity.class);
        intent.putExtra("NUMBER1", excellent_tip);
        intent.putExtra("NUMBER2", average_tip);
        intent.putExtra("NUMBER3", bad_tip );
        startActivityForResult(intent, 999);
    }


    public void onRadioButtonClicked(View view) {
        boolean checked = ((RadioButton) view).isChecked();
        float bill;

        // Check which radio button was clicked
        switch (view.getId()) {
            case R.id.excellent_button:
                if (checked) {
                    EditText b = (EditText)findViewById(R.id.bill);
                    if (b.getText().toString().equals(""))
                        bill = 0;
                    else bill = Float.parseFloat(b.getText().toString());
                    compute_tip(bill, excellent_tip);
                }
                break;
            case R.id.average_button:
                if (checked) {
                    EditText b = (EditText)findViewById(R.id.bill);
                    if (b.getText().toString().equals(""))
                        bill = 0;
                    else bill = Float.parseFloat(b.getText().toString());
                    compute_tip(bill, average_tip);
                }
                break;
            case R.id.lacking_button:
                if (checked) {
                    EditText b = (EditText)findViewById(R.id.bill);
                    if (b.getText().toString().equals(""))
                        bill = 0;
                    else bill = Float.parseFloat(b.getText().toString());
                    compute_tip(bill, bad_tip);
                }

                break;
        }
    }
    public static String roundToTwoDigit(float paramFloat) {
        return String.format("%.2f%n", paramFloat);
    }
    void compute_tip(float bill, int percent) {
        float pct= (float)percent/100;
        float tip = bill * pct;
        float total = bill + tip;
        TextView t = (TextView)findViewById(R.id.computed_tip);
        String s = roundToTwoDigit(tip);
        t.setText(s);
        t = (TextView)findViewById(R.id.bill_total);
        s = roundToTwoDigit(total);
        t.setText(s);

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(requestCode == 999 && resultCode == RESULT_OK){
            excellent_tip = data.getIntExtra("NUMBER1", excellent_tip);
            average_tip= data.getIntExtra("NUMBER2", average_tip);
            bad_tip = data.getIntExtra("NUMBER3", bad_tip);
        }

    }
}

File: Lab7krimal/app/src/main/java/fall2020/mobileapp/lab7_krimal/MainActivity.kt
package fall2020.mobileapp.lab7_krimal

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.RadioButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    var excellent_tip = 20
    var average_tip = 18
    var bad_tip = 14
    var button: Button? = null
    var Lab1Button: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button =findViewById<View>(R.id.secondActivityButton) as Button
        Lab1Button = findViewById<View>(R.id.Implicit_call) as Button


        button!!.setOnClickListener { openNewActivity() }

        Lab1Button!!.setOnClickListener { Lab1Activity() }
    }


    fun Lab1Activity() {
        val intent = Intent()
        intent.action = "android.intent.action.VIEW"
        intent.addCategory("android.intent.category.DEFAULT")
        startActivity(intent)
    }

    fun openNewActivity() {
        val intent = Intent(this, NewActivity::class.java)
        intent.putExtra("NUMBER1", excellent_tip)
        intent.putExtra("NUMBER2", average_tip)
        intent.putExtra("NUMBER3", bad_tip)
        startActivityForResult(intent, 999)
    }

    fun onRadioButtonClicked(view: View) {
        val checked = (view as RadioButton).isChecked
        val bill: Float
        when (view.getId()) {
            R.id.excellent_button -> if (checked) {
                val b = findViewById<View>(R.id.bill) as EditText
                bill = if (b.text.toString() == "") 0f else b.text.toString().toFloat()
                compute_tip(bill, excellent_tip)
            }
            R.id.average_button -> if (checked) {
                val b = findViewById<View>(R.id.bill) as EditText
                bill = if (b.text.toString() == "") 0f else b.text.toString().toFloat()
                compute_tip(bill, average_tip)
            }
            R.id.lacking_button -> if (checked) {
                val b = findViewById<View>(R.id.bill) as EditText
                bill = if (b.text.toString() == "") 0f else b.text.toString().toFloat()
                compute_tip(bill, bad_tip)
            }
        }
    }
    fun roundToTwoDigit(paramFloat: Float): String? {
        return String.format("%.2f%n", paramFloat)
    }

    fun compute_tip(bill: Float, percent: Int) {
        val pct = percent.toFloat() / 100
        val tip = bill * pct
        val total = bill + tip
        var t = findViewById<View>(R.id.computed_tip) as TextView
        var s = roundToTwoDigit(tip)
        t.text = s
        t = findViewById<View>(R.id.bill_total) as TextView
        s = roundToTwoDigit(total)
        t.text = s
    }

    override fun onActivityResult(
        requestCode: Int,
        resultCode: Int,
        data: Intent?
    ) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == 999 && resultCode == Activity.RESULT_OK) {
            if(data != null) {
                excellent_tip = data.getIntExtra("NUMBER1", excellent_tip)
                average_tip = data.getIntExtra("NUMBER2", average_tip)
                bad_tip = data.getIntExtra("NUMBER3", bad_tip)
            }
        }
    }

}
